controllers/updateservice_controller.go

```diff
@@ -674,8 +690,54 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 	return fmt.Errorf(fmt.Sprintf("UpdateService route name %q %s Route name %s", routeName, errReasons[0], errReasons[1]))
 }
 
-func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
+func (r *UpdateServiceReconciler) ensureMetadataRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
+	route := resources.metadataRoute
+	foundRoute := &routev1.Route{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: route.Name, Namespace: route.Namespace}, foundRoute)
+	if err != nil && apiErrors.IsNotFound(err) {
+		// Set UpdateService instance as the owner and controller
+		if err = controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
+			return err
+		}
+		reqLogger.Info("Creating Route", "Namespace", route.Namespace, "Name", route.Name)
+		if err = r.Client.Create(ctx, route); err != nil {
+			handleErr(reqLogger, &instance.Status, "CreateRouteFailed", err)
+		}
+		return err
+	} else if err != nil {
+		handleErr(reqLogger, &instance.Status, "GetRouteFailed", err)
+		return err
+	}
 
+	if uri, _, err := routeapihelpers.IngressURI(foundRoute, ""); err == nil {
+		instance.Status.MetadataURI = uri.String()
+	} else {
+		handleErr(reqLogger, &instance.Status, "RouteIngressFailed", err)
+	}
+
+	updated := foundRoute.DeepCopy()
+	// Keep found tls for later use
+	tls := updated.Spec.TLS
+	// This is just so we compare the Spec on the two objects but make an exception for Spec.TLS
+	updated.Spec.TLS = route.Spec.TLS
+
+	// found existing resource; let's compare and update if needed
+	if !reflect.DeepEqual(updated.Spec, route.Spec) {
```

Comment:
I'd expect controllers to use semantic.DeepEqual from apimachinery? https://github.com/kubernetes/apimachinery/blob/master/pkg/api/equality/semantic.go

But probably not a big deal in this case.

Reply:
I'm copy/pasting existing patterns, so I'll punt this pivot to follow-up work.

Reply:
I would prefer semantic.DeepEqual from apimachinery as the reason semantic.DeepEqual exists in apimachinery. I would guess that it is more future proof